openapi-spec/swagger.yaml

```diff
@@ -3003,3 +3117,80 @@ components:
                   - nvmeof
 
       description: Host is a consumer of volume from storage.
+    MetricSpec:
+      description: >-
+        Metric spec represents metrics of diifent type, which has name, instanceID label, timestamp , value etc..
+      allOf:
+        - $ref: '#/components/schemas/BaseModel'
+        - type: object
+          required:
+            - name
+          properties:
+            instanceID:
+              type: string
+              readOnly: true
+            instanceName:
+              type: string
+              readOnly: true
+            job:
+              type: string
+            name:
+              type: string
+              default: default
+            unit:
+              type: string
+              readOnly: true
+            aggrType:
+              type: string
+            metricValues:
+              type: array
+              description: Metric values for the timestamp
+              items:
+                type: object
+                properties:
+                  timestamp:
+                    type: integer
+                    format: int64
+                    description: Timestamp of the metric
+                    example: 1575363783
+                  value:
+                    type: number
+                    description: Vlue of the metric for this timestamp
+                    example: 16.74
+            labels:
+              type: object
+              additionalProperties:
+                type: string
+              example:
+                key1: value1
+                key2: value2
+    CollectMetricSpec:
+      description: >-
+        Collect Metric spec specifies the driver to trigger collect metrics action.
+      allOf:
+        - $ref: '#/components/schemas/BaseModel'
+        - type: object
+          required:
+            - driverType
+          properties:
+            driverType:
+              type: string
```

Comment:
which driver types does it support? maybe enum is better here

Reply:
lvm, ceph and huawei_oceanstor. enum provided for this.